posydon/binary_evol/SN/step_SN.py

```diff
@@ -496,29 +497,60 @@ def collapse_star(self, star):
                             print('matched to model:', tmp)
                         MODEL_NAME_SEL = tmp
                 
+                
+                
                 # check if selected MODEL is supported
                 if MODEL_NAME_SEL is None:
                     raise ValueError('Your model assumptions are not'
                                      'supported!')   
                 elif getattr(star, MODEL_NAME_SEL) is None:
                     # NOTE: this option is needed to do the collapse
                     # for stars evolved with the step_detached or 
-                    # steo_disrupted.
+                    # step_disrupted.
                     # allow to continue with the collapse with profile
                     # or core masses
                     warnings.warn(f'{MODEL_NAME_SEL}: The collapsed star '
                                      'was not interpolated! If use_profiles '
                                      'or use_core_masses is set to True, '
                                      'continue with the collapse.')                 
                 else:
+                    # store some properties of the star object
+                    # to be used for collapse verification
+                    pre_SN_c_core_mass = star.c_core_mass
+                    pre_SN_he_core_mass = star.he_core_mass
+                    pre_SN_mass = star.mass
+
                     MODEL_properties = getattr(star, MODEL_NAME_SEL)
                     for key, value in MODEL_properties.items():
                         setattr(star, key, value)
-                
+
                     for key in STARPROPERTIES:
                         if key not in ["state", "mass", "spin",
                                         "m_disk_accreted ", "m_disk_radiated"]:
                             setattr(star, key, None)
+                        
+                    # check if SN_type matches the predicted CO
+                    # and force the SN_type to match the predicted CO.
+                    # ie WD is no SN
+                    if star.state == 'WD' and star.SN_type != "WD":
+                        star.SN_type = 'WD'
```

Comment:
Can we add also a warning message, including the old, erroneous SN_type, so that the user knows that the something may have happenned and that the state and SNtype were forced to be the same.